### ISSUE TITLE
ShapeletModel extensions

### ISSUE THREAD

**Issue by GillesVandewiele** [tslearn-team/tslearn#120]
The following stuff would be very interesting to implement/integrate in the current version of ShapeletModel.

1) Remove all logics for the binary case, and always use a "softmax" activation in the output layer (even for binary case).
--> `predict_proba()` would just be `model.predict()`
--> `predict` would be `predict_proba().argmax(axis=1)`

2) Making ShapeletModel serializable/picklable and removing SerializableShapeletModel afterwards.
--> This will require patching of some keras functions. We need to make sure we do not need to use a very new version in order to have backward compatibility.

3) During the refactoring to make all estimators sklearn-compliant, we noticed weird issues on a simple dataset with a low learning rate. The dataset should be very easy to classify, but the model achieved 0% accuracy, which is much worse than random. It seems like it is learning the opposite of what it should learn.
```
dataset:
----------
random_walk_blobs(n_ts_per_blob=100, random_state=42,
                                 n_blobs=3, noise_level=0.1, sz=100)

learning rate:
-----------------
K.set_value(self.model.optimizer.lr, 0.001)
```

4) Implement #118 and create an example notebook with some plot.

**Comment by GillesVandewiele** [comment]
We have also been noticing some weird convergence issues in the patched `check_classifiers_train` test for both shapelet models that should be looked at. (See PR #116 )

**Comment by GillesVandewiele** [comment]
Related to this are the following issues: #96 , #85 , and #53 

@rtavenar this is something I could do when my next PR is finished.

**Comment by rtavenar** [comment]
OK, great!

Once again, I disagree with removing all logics that are specific to the binary case, since they correspond to what is written in the original paper, and I think we should stick to implementing what is in the paper.

But I fully agree that it is a mess at the moment and should be made clearer. Thanks for taking care of that @GillesVandewiele

**Comment by rtavenar** [comment]
@GillesVandewiele could you check which of these items are still on our todo list ? It seems some of them are outdated,at least.

**Comment by GillesVandewiele** [comment]
It indeed looks like most of those points are handled, except for the SerializableShapeletModel (here I am again :) ). You did indeed mention in a previous PR that there are some issues, but I remember that we were able to fix these some time ago, but that it required a too recent version of tensorflow.

I still have to take a look whether it can actually be fixed. It would allow us to remove a significant chunk of code that does not have that much added value.

**Comment by rtavenar** [comment]
Ok I agree, so we should maybe first move to TF2 and check if it helps for cloning. It would also allow full support for Python3.8
